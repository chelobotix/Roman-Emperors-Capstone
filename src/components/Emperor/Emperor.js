import { useParams } from "react-router-dom";
import { useSelector } from "react-redux";
import style from "./Emperor.module.css";

const Emperor = () => {
  const state = useSelector((state) => state.emperors);
  const { name } = useParams();
  const emperorDetails = state.emperorsArray.records.filter((data) => data.fields.name === name);

  return (
    <div className={style.emperorContainer}>
      <h2 className={style.h2}>Emperor {emperorDetails[0].fields.name}</h2>
      <img src={emperorDetails[0].fields.image} alt={name} className={style.img} />
      <ul className={style.ul}>
        <li className={style.li}>
          <p className={style.p1}>Full Name</p>
          <p className={style.p2}>{emperorDetails[0].fields.name_full}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Reign Start</p>
          <p className={style.p2}>{emperorDetails[0].fields.reign_start}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Reign End</p>
          <p className={style.p2}>{emperorDetails[0].fields.reign_end}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Birth City</p>
          <p className={style.p2}>{emperorDetails[0].fields.birth_cty}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Birth Province</p>
          <p className={style.p2}>{emperorDetails[0].fields.birth_prv}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Era</p>
          <p className={style.p2}>{emperorDetails[0].fields.era}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Date of Dead</p>
          <p className={style.p2}>{emperorDetails[0].fields.death}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Killer</p>
          <p className={style.p2}>{emperorDetails[0].fields.killer}</p>
        </li>
        <li className={style.li}>
          <p className={style.p1}>Cause of Dead</p>
          <p className={style.p2}>{emperorDetails[0].fields.cause}</p>
        </li>
      </ul>
    </div>
  );
};

export default Emperor;
