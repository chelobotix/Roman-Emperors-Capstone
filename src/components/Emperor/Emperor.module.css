.emperorContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #4369b0;
}

.h2 {
  font-size: 1.5rem;
  color: #fff;
  text-decoration: underline;
  margin-bottom: 10px;
}

.img {
  width: 150px;
  border-radius: 20%;
}

.li {
  padding: 10px 5px;
}

.p1 {
  background-color: #4369b0;
  color: #c2c2c2;
  font-style: italic;
}

.p2 {
  background-color: #3d60a2;
  color: #fff;
}
