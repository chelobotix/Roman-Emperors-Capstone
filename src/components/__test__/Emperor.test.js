import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import Emperor from '../Emperor/Emperor';

const emperorsArray = {
  records: [
    {
      fields:
      {
        name: 'Augustus',
        mission_name: 'Mission test',
        description: 'Description Test',
        reserved: false,
      },
    }
  ]
};

const mockStore = configureStore([]);
const initialState = {
  emperors: {
    emperorsArray: emperorsArray,
  },
};

const store = mockStore(initialState);

jest.mock('react-router-dom', () => ({
  useParams: () => ({ name: 'Augustus' }),
}));

describe('Emperor', () => {
  test('Check if render', () => {
    render(
      <Provider store={store}>
        <Emperor />
      </Provider>,
    );
    const ul = screen.getByRole('list');
    expect(ul).toBeInTheDocument();
  });

  test('Home Tree', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Emperor />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
