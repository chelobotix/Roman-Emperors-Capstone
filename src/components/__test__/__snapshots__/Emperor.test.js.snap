// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emperor Home Tree 1`] = `
<div
  className="emperorContainer"
>
  <h2
    className="h2"
  >
    Emperor 
    Augustus
  </h2>
  <img
    alt="Augustus"
    className="img"
  />
  <ul
    className="ul"
  >
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Full Name
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Reign Start
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Reign End
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Birth City
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Birth Province
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Era
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Date of Dead
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Killer
      </p>
      <p
        className="p2"
      />
    </li>
    <li
      className="li"
    >
      <p
        className="p1"
      >
        Cause of Dead
      </p>
      <p
        className="p2"
      />
    </li>
  </ul>
</div>
`;
